### DISCUSSION TITLE
Stop `save_hyperparameters` from logging anything`

### DISCUSSION THREAD

**Discussion by sachinruk** [Lightning-AI/pytorch-lightning#11052] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hi, just wondering if there was to stop `save_hyperparameters` altogether? The workaround I'm thinking of was to use the `inspect` module and getting all the arguments in `ignore` parameter. The [SO discussion is here](https://stackoverflow.com/questions/70343027/get-argument-list-from-within-init).

**Comment by rohitgr7** [comment]
Hey,

didn't get you here. Do you want to stop `save_hyperparameters` and not log the hparams to the logger or ignore all params from `save_hyperparameters`?

**Comment by sachinruk** [comment]
That is correct.

On Tue, 14 Dec 2021, 10:26 pm Rohit Gupta, ***@***.***> wrote:

> Hey,
>
> didn't get you here. Do you want to stop save_hyperparameters and not log
> the hparams to the logger to ignore all params from save_hyperparameters?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/PyTorchLightning/pytorch-lightning/discussions/11052#discussioncomment-1804496>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAKYO3YS65PBVMVSNWECNCTUQ4SWNANCNFSM5J7WOZWQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

-- 
**
** <https://www.canva.com/>Empowering the world to design
Share accurate 
information on COVID-19 and spread messages of support to your community.
Here are some resources 
<https://about.canva.com/coronavirus-awareness-collection/?utm_medium=pr&utm_source=news&utm_campaign=covid19_templates> 
that can help.
 <https://twitter.com/canva> <https://facebook.com/canva> 
<https://au.linkedin.com/company/canva> <https://twitter.com/canva>  
<https://facebook.com/canva>  <https://au.linkedin.com/company/canva>  
<https://instagram.com/canva>

**Reply by rohitgr7** [reply]
you can override logger's `log_hyperparams` method and disable the upload there. then pass it in the trainer:
```py
trainer = Trainer(logger=CustomLogger(...))
```
